flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/DynamicKafkaSerializationSchema.java

```diff
@@ -120,13 +119,17 @@ public void open(SerializationSchema.InitializationContext context) throws Excep
         if (keySerialization == null) {
```

Comment:
Why do we still need this class?

Reply:
As long as we use the `FlinkKafkaProducer` we have to support the old schema.

Reply:
Please get rid of it in your next PR.